src/Persistence/MassTransit.EntityFrameworkCoreIntegration/EntityFrameworkCoreIntegration/BusOutboxDeliveryService.cs

```diff
@@ -115,10 +115,10 @@ async Task<int> Execute()
 
                     try
                     {
-                        var outboxState = await dbContext.Set<OutboxState>()
+                        var outboxStateList = await dbContext.Set<OutboxState>()
                             .FromSqlRaw(_getOutboxIdStatement)
-                            .AsTracking()
-                            .SingleOrDefaultAsync(timeoutToken.Token).ConfigureAwait(false);
+                            .AsTracking().ToListAsync(timeoutToken.Token).ConfigureAwait(false);
+                        var outboxState = outboxStateList.SingleOrDefault();
```

Comment:
Why is this change necessary?

Reply:
When the ToList() is used, then the statement is not passed as a subquery.

1. With SingleOrDefault this looks:

```
select x.col1, x.col2 ... x.coln
from
(
   select ... from OutboxState // _getOutboxIdStatement
)
```

2. With ToList() there is no subselect so the FOR UPDATE SKIP LOCKED can be used in oracle formatter CreateOutboxStatement